### DISCUSSION TITLE
classes/class_progressbar

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#47] [User-contributed notes]
# classes/class_progressbar

Inherits: Range< Control< CanvasItem< Node< Object A control used for visual representation of a percentage. Description: A control used for visual representation of a percentage. Shows fill percen...

https://docs.godotengine.org/en/stable/classes/class_progressbar.html

**Comment by TheSuchaVoda** [comment]
I recommend making the ProgressBar.Size big (and the ProgressBar.Scale small depending on how big you want it to be) to make the progress color not "jump" as much

**Reply by Calinou** [reply]
Scaling Control nodes should be avoided unless it's for animation purposes, as scaling will break within containers.

I also don't understand why the animation would look smoother with a large size and a small scale. As far as I know, ProgressBar precision is based on the Range properties (**Step** in this case).

Are you referring to the **Snap Controls to Pixels** project setting? You can disable it, but it'll make UI rendering look less crisp overall. This would perhaps need a way to control it on a per-node basis, or disable it for progress bar rendering only so it can be smoother.

**Reply by TheSuchaVoda** [reply]
Lets say I have a progress bar with size 8x1 = then the fill jumps one by one pixel instead of filling smoothly (so i made it 800x100 with scale 0.01x0.01).

But I am a beginner, I didnt know about this container problem and also the Snap Controls to Pixels setting, but now that I tried disabling it, it didnt help, and it still was filling up one pixel at a time.

Maybe I did do something wrong, but I dont think so.

**Comment by KosoEPK** [comment]
I feel like there should be a way to get the percentage shown and know how it is calculated

**Reply by Calinou** [reply]
The formula used to calculate the percentage displayed is here:

https://github.com/godotengine/godot/blob/7b1ed520bdcca21d36e25e2094802b9b33dae2c6/scene/gui/progress_bar.cpp#L140-L165

In most situations, all you need is `int((value - min) / (max - min)) * 100`.

**Reply by setnoset** [reply]
You will probably have to place a label somewhere in the progress bar to show the percentage.

**Reply by setnoset** [reply]
oops, I didn't read. there is a built in option for the in the bar.

**Reply by KosoEPK** [reply]
For some reason my progress bar goes from 59% to 57% when the value is 58, max being 100 and min being 0, there is probably a mistake in my code i guess

**Comment by Heistergand** [comment]
I feel like there should be a show_value to show the current value at the progressbar as alternative to the show_percentage.
